sphinx_gallery/gen_gallery.py

```diff
@@ -180,11 +182,22 @@ def generate_gallery_rst(app):
 
         computation_times += this_computation_times
 
+        key = None
+        if gallery_conf['examples_dirs_order'] is not None:
+            base_examples_dir = os.path.basename(examples_dir)
+            ordering = gallery_conf['examples_dirs_order']
+            if not isinstance(ordering, dict):
+                raise ValueError('If not None, examples_dirs_order'
+                                 ' must be a dictionary of folder:'
+                                 ' ordering pairs')
+            if base_examples_dir in ordering.keys():
+                key = ExplicitOrderKey(ordering[base_examples_dir])
+
```

Comment:
Once our default is the empty dictionary we don't need the first check. Directly start checking the user has input a dictionary the config or raise the ValueError. The message of the error is unclear to me, we need to make this more comprehensible.